### DISCUSSION TITLE
Is possible updating Model (with Sushi) for realtime?

### DISCUSSION THREAD

**Discussion by swiftalker** [filamentphp/filament#7898] [General]
hi everybody,

I want to ask about the table on filamentphp, I have a scenario where I use the filament form to call the API, for that I use afterStateUpdated, then after I call the API, I want to release it in the table, of course this can't be done directly through the filamentphp table, but there is some recommendation where they use caleb porzio's Laravel Sushi, after that i used it, but there is a problem here, because the data i fetch is from API, it can't be received via model directly so i use getter and setter method, where i set $rows on sushi then getRows can be called with the data I want, ok after the scenario all goes, and getRows on sushi is perfectly what I want, now the problem is the table when afterStateUpdate doesn't change at all, even the data is called in full, this is the problem that I'm facing right now, so that this table can reload data after afterStateUpdate and after the API is called, what should I do?

**Comment by leandrocfe** [comment]
[How to consume an external API with Filament Tables](https://filamentphp.com/community/how-to-consume-an-external-api-with-filament-tables)
Are resources being utilized? In case they involve pages, saving will lead to a redirection to the index page

**Reply by swiftalker** [reply]
Thank you, previously I have read your article regarding my problem, but this has not been a solution in my case.

Yes, resource has been utilized. This case involved the pages also, actually custom pages, more clearly maybe I will provide a code reference.

For the pages
```
class ApiGetServices extends Page implements Forms\Contracts\HasForms, Tables\Contracts\HasTable
{
    use Forms\Concerns\InteractsWithForms; 
    use Tables\Concerns\InteractsWithTable; 

    public $api;
    public ?array $data = [];
    public $apiData = [];

    protected static string $resource = ApiMapperResource::class;

    protected static string $view = 'x';
    
    public function mount($id = null) {
        $this->form->fill();
        $this->api = $this->getModel()::find($id);
    }

    public function getData($state)
    {
        if ($state == null) {
            return;
        }
        $temporary = new TemporaryService;
        $services = new \App\Api\Main;
        $api = $this->api;
        $temporary->setRows($services->getServices($api));
        $this->apiData = $temporary->getRows();
        return $this->apiData;
    }

    public function form(Forms\Form $form): Forms\Form
    {
        return $form
            ->schema([
                Forms\Components\Select::make('apiProvider')
                ->live()
                ->label('API Provider')
                ->options(
                    $this->getModel()::where('status->api_active', 1)->orderByDesc('id')->pluck('name', 'id')
                )
                ->afterStateUpdated(function (?string $state, ?string $old, ) {
                    $this->getData($state);
                })
            ])
            ->model($this->getModel())
            ->statePath('data');
    }

    public function getEloquentQuery(): Builder
    {
        return TemporaryService::query();
    }
    public function table(Tables\Table $table): Tables\Table
    {
        return $table
            ->query($this->getEloquentQuery())
            ->columns([
                Columns\TextColumn::make('name')->label('Name Service')
            ]);
    }
}
```

for the model
```

class TemporaryService extends Model
{
    use \Sushi\Sushi;
    protected $connection = 'sqlite';

    protected $rows = [];

    public function setRows($data = []) {
        $this->rows = $data;
    }
    
    public function getRows() {
        return $this->rows;
    }

    protected function sushiShouldCache()
    {
        return true;
    }
}
```

something like this is an overview of my code, everything works, only when this table becomes a problem, the data doesn't appear at all after the change event.

**Reply by leandrocfe** [reply]
did you try to remove this?
```php
protected function sushiShouldCache()
{
    return true;
}
```

**Reply by swiftalker** [reply]
Yes, i was remove shouldCache and still doesnt work.

**Reply by CoringaWc** [reply]
$temporary->setRows($services->getServices($api));
 $temporary->migrate();
